schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -592,6 +592,21 @@ public SchemaVersionInfo getSchemaVersionInfo(SchemaVersionKey schemaVersionKey)
         return schemaVersionInfoCache.getSchema(SchemaVersionInfoCache.Key.of(schemaVersionKey));
     }
 
+    @Override
+    public void deleteSchemaVersion(SchemaVersionKey schemaVersionKey) throws SchemaNotFoundException {
+        synchronized (addOrUpdateLock) {
+            schemaVersionInfoCache.invalidateSchema(new SchemaVersionInfoCache.Key(schemaVersionKey));
```

Comment:
You may want to retrieve from cache and then invalidate that like below. This will avoid hitting storage layer if it exists in cache.

```
SchemaVersionInfoCache.Key key = new SchemaVersionInfoCache.Key(schemaVersionKey);
SchemaVersionInfo schemaVersionInfo = schemaVersionInfoCache.getSchema(key);
schemaVersionInfoCache.invalidateSchema(key);
```